evalml/tests/component_tests/test_time_series_regularizer.py

```diff
@@ -0,0 +1,267 @@
+        X, y, X_output, y_output, error_dict, has_target=False
+    )
+
+
+@pytest.mark.parametrize(
+    "dataset", ["beginning", "middle", "end", "scattered", "continuous"]
+)
+def test_ts_regularizer_duplicate(
+    dataset,
+    duplicate_beginning,
+    duplicate_middle,
+    duplicate_end,
+    duplicate_scattered,
+    duplicate_continuous,
+):
+
+    if dataset == "beginning":
+        dates = duplicate_beginning
+    elif dataset == "middle":
+        dates = duplicate_middle
+    elif dataset == "end":
+        dates = duplicate_end
+    elif dataset == "scattered":
+        dates = duplicate_scattered
+    else:
+        dates = duplicate_continuous
+
+    X, y = get_df(dates)
+
+    ts_regularizer = TimeSeriesRegularizer(time_index="dates")
+    X_output, y_output = ts_regularizer.fit_transform(X, y)
+
+    error_dict = ts_regularizer.error_dict
+    assert_features_and_length_equal(X, y, X_output, y_output, error_dict)
+
+
+@pytest.mark.parametrize(
+    "dataset", ["beginning", "middle", "end", "scattered", "continuous"]
+)
+def test_ts_regularizer_missing(
+    dataset,
+    missing_beginning,
+    missing_middle,
+    missing_end,
+    missing_scattered,
+    missing_continuous,
+):
+
+    if dataset == "beginning":
+        dates = missing_beginning
+    elif dataset == "middle":
+        dates = missing_middle
+    elif dataset == "end":
+        dates = missing_end
+    elif dataset == "scattered":
+        dates = missing_scattered
+    else:
+        dates = missing_continuous
+
+    X, y = get_df(dates)
+
+    ts_regularizer = TimeSeriesRegularizer(time_index="dates")
+    X_output, y_output = ts_regularizer.fit_transform(X, y)
+
+    error_dict = ts_regularizer.error_dict
+    assert_features_and_length_equal(X, y, X_output, y_output, error_dict)
+
+
+@pytest.mark.parametrize(
+    "dataset", ["beginning", "middle", "end", "scattered", "continuous"]
+)
+def test_ts_regularizer_uneven(
+    dataset,
+    uneven_beginning,
+    uneven_middle,
+    uneven_end,
+    uneven_scattered,
+    uneven_continuous,
+):
+
+    if dataset == "beginning":
+        dates = uneven_beginning
+    elif dataset == "middle":
+        dates = uneven_middle
+    elif dataset == "end":
+        dates = uneven_end
+    elif dataset == "scattered":
+        dates = uneven_scattered
+    else:
+        dates = uneven_continuous
+
+    X, y = get_df(dates)
+
+    ts_regularizer = TimeSeriesRegularizer(time_index="dates")
+    X_output, y_output = ts_regularizer.fit_transform(X, y)
+
+    if dataset == "beginning":
+        assert X.iloc[0]["dates"] not in X_output["dates"]
+        assert y.iloc[0] not in y_output.values
+    elif dataset == "end":
+        assert X.iloc[-1]["dates"] not in X_output["dates"]
+        assert y.iloc[-1] not in y_output.values
+
+    error_dict = ts_regularizer.error_dict
+    assert_features_and_length_equal(X, y, X_output, y_output, error_dict)
+
+
+def test_ts_regularizer_combination(combination):
```

Comment:
Maybe just parametrize one of your two combination tests over a "X only" parameter?